### ISSUE TITLE
Create custom theme for Backstage

### ISSUE THREAD

**Issue by rtaniwa** [janus-idp/backstage-showcase#506]
## Goal

Backstage is a user interface, heavy product. Therefore, the experience of the product is directly correlated with the success and experience found on the front end. We want to differentiate our look and feel from that of the upstream Backstage, however making significant changes to the user experience of an open-source project is challenging and will take extensive collaboration with the community. In the meantime, we must still find a way to differentiate Janus from backstage visually.

Solution

We can take advantage of a feature built in the backstage called scenes. This enables us to change the look and feel of mini components backstage without hindering compatibility. This means that plug-ins and other basic components from the Backstage community will be fully compatible with these changes, and no changes to those plug-ins will be required for the theme to be supported. 

Resource

[How to create a custom theme](https://backstage.io/docs/getting-started/app-custom-theme)
## Acceptance criteria

- tbd

## Requirements

- [ ] Test plan
- [ ] Documentation

## Issues in Epic

- [x] https://github.com/janus-idp/backstage-showcase/issues/532
- [x] https://github.com/janus-idp/backstage-showcase/issues/533
- [ ] https://github.com/janus-idp/backstage-showcase/issues/534
- [ ] https://github.com/janus-idp/backstage-showcase/issues/535
- [ ] https://github.com/janus-idp/backstage-showcase/issues/536
- [ ] https://github.com/janus-idp/backstage-showcase/issues/537
- [ ] https://github.com/janus-idp/backstage-showcase/issues/538
- [ ] https://github.com/janus-idp/backstage-showcase/issues/823

## Notes

**Additional context**
Add any other context or screenshots about the epic here.

**Comment by gashcrumb** [comment]
Some initial tasks on this one in no particular order, I'll put them here and decide if issues need to be created for them as needed.

- Investigate upstream backstage storybook and how a theme could be loaded there, this would ensure complete coverage and make theme development much easier.
- Consider how to adjust the current approach to cater for a possible dark variant of the theme
- Consider making the theme like a base Patternfly material theme that could be then extended from for branding purposes
- Get the theme contributed somewhere, figure out where that should be

**Comment by invincibleJai** [comment]
> Some initial tasks on this one in no particular order, I'll put them here and decide if issues need to be created for them as needed.
> 
> * Investigate upstream backstage storybook and how a theme could be loaded there, this would ensure complete coverage and make theme development much easier.
>   
>   * @invincibleJai suggested looking at adding support in the upstream storybook to load themes
>   * Upstream also has better access to some wrapper functions that should make this effort easier
> * Consider how to adjust the current approach to cater for a possible dark variant of the theme
> * ~Consider making the theme like a base Patternfly material theme that could be then extended from for branding purposes~ - in the near and mid term Backstage is probably going to continue to support Material v4 & v5 which rules out a pure Material v5 theme at least for now.
> * Get the theme contributed somewhere, figure out where that should be.  For now it's [here](https://github.com/gashcrumb/patternfly-backstage-theme).
> * ~Consider theme development setup as a standalone package and how it could be incorporated into backstage-plugins, possibly within [this monorepo](https://github.com/janus-idp/backstage-plugins/tree/main/packages) unless it'd make sense to turn it into a plugin as well.~ - suggestion above to look into adding upstream support for theme development would be a better strategy for ongoing maintenance.
> * Upgrade upstream storybook to v7, there's a ticket and some effort upstream already to start from.

Thanks @gashcrumb, let's sync up and we need to create stories if they are independent then can be worked in parallel

**Comment by gashcrumb** [comment]
Parked this for now here -> https://github.com/gashcrumb/backstage-showcase/commit/ff04ffa6ab3f2827225746e8b810eebe3554cf7a

**Comment by rtaniwa** [comment]
Can we get rid of this issue now that we have one tracking this work in Jira?

**Comment by gashcrumb** [comment]
👍
